### PR TITLE
Support for Excel files

### LINKED ISSUE #1103: Excel File Support
### 🚀 The feature

I need excel file support!

### Motivation, pitch

ALot

### PR DESCRIPTION
**Pull request by Dev-Khant**
## Description

Added support for excel files.

Fixes #1103

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## How Has This Been Tested?

- [x] Unit Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by Dev-Khant** [comment]
@deshraj Can you please review this?

File: pyproject.toml [hunk 1]
```diff
@@ -123,6 +123,8 @@ cohere = { version = "^4.27", optional = true }
 together = { version = "^0.2.8", optional = true }
 weaviate-client = { version = "^3.24.1", optional = true }
 docx2txt = { version = "^0.8", optional = true }
+openpyxl = { version = "^3.1.2", optional = true }
+xlrd = { version = "^2.0.1", optional = true }
```
Comment (deshraj) [inline pyproject.toml:127]: Going forward, we are planning to not add everything to the optional dependencies. Instead, can you please:

1. Add instructions in the docs to install these two packages
2. Try to import these on the loader file and if import fails, show the exception to install these packages.

Comment (Dev-Khant) [reply pyproject.toml]: @deshraj I have made the change, please check

**Comment by Dev-Khant** [comment]
@deshraj Tests are failing because of Qdrant vectordb

**Comment by Dev-Khant** [comment]
Hi @deshraj Can you please review it?

**Comment by Dev-Khant** [comment]
@deshraj @taranjeet Please can you review this?

**Review by deshraj — APPROVED** [review]

**Pull request merged**